src/mam/mod_mam_riak_timed_arch_yz.erl

```diff
@@ -231,6 +249,10 @@ create_obj(Host, MsgId, SourceJID, Packet, Type) ->
             fun(R) -> riakc_register:set(MsgId, R) end},
            {{<<"source_jid">>, register},
             fun(R) -> riakc_register:set(SourceJID, R) end},
+           {{<<"owner_jid">>, register},
```

Comment:
`owner_jid` is kind of ambiguous in MAM context, because it may mean both "message owner" or "archive owner". I think that we should come up with a better name, like `sender_jid` or `real_sender_jid`.

Reply:
The problem here is that, this is not "sender's" JID. In case of MUC, yeah, sure. But for PM this archive owner. Would `archive_owner_jid` be better? I guess it's not "correct" for MUC in such case. I feel that, there should be a different "schema" for PM and MUC, it would be much less confusing...

Reply:
Actually `msg_owner_jid` feels quite nice. @fenek What do you think?

Reply:
Yup, `msg_owner_jid` will be accurate IMO. :)